text/3838-scalable-vectors.md

```diff
@@ -0,0 +1,1271 @@
+┌───────────────────────┬───────────────────────┬───────────────────────┐
+│ f32 │ f32 │ f32 │ f32 │ f32 │ f32 │ f32 │ f32 │ f32 │ f32 │ f32 │ f32 │ ← `svfloat32_t`
+└───────────────────────┴───────────────────────┴───────────────────────┘
+```
+
+The type marker field is solely used to get the element type for the codegen
+backend. It must one of the following types:
+
+- `u8`, `u16`, `u32` or `u64`
+- `i8`, `i16`, `i32` or `i64`
+- `f16`, `f32` or `f64`
+- `bool`
+
+It is not permitted to project into scalable vector types and access the type
+marker field.
+
+## Tuples of vectors
+[tuples-of-vectors]: #tuples-of-vectors
+
+Structs of scalable vectors are supported, but every element of the struct must
+have the same scalable vector type. This will enable definition of "tuple of
+vector" types, such as `svfloat32x2_t` below, that are used in some load and
+store intrinsics.
+
+```rust
+#[rustc_scalable_vector]
+pub struct svfloat32x2_t(svfloat32_t, svfloat32_t);
+```
+
+```text
+◁───────────── vscale x f32 x 4 ─────────────▷ ◁────── f32 x 4 ──────▷
+┌──────────────────────────────────────────────┬───────────────────────┐  ┐
+│                      ...                     │ f32 │ f32 │ f32 │ f32 │  │
+└──────────────────────────────────────────────┴───────────────────────┘  ├─ svfloat32x2_t
+┌──────────────────────────────────────────────┬───────────────────────┐  │
+│                      ...                     │ f32 │ f32 │ f32 │ f32 │  │
+└──────────────────────────────────────────────┴───────────────────────┘  ┘
+```
+
+Structs must be still be annotated with `#[rustc_scalable_vector]`, so end-users
+cannot define their own structs of scalable vectors. It is not permitted to
+project into structs and access the individual vectors.
+
+## Properties of scalable vectors
+[properties-of-scalable-vector-types]: #properties-of-scalable-vectors
+
+Scalable vectors are necessarily non-`const Sized` (from [rfcs#3729]) as they
+behave like value types but the exact size cannot be known at compilation time.
+
+[rfcs#3729] allows these types to implement `Clone` (and consequently `Copy`) as
+`Clone` only requires an implementation of `Sized`, irrespective of constness.
+
+Scalable vector types have some further restrictions due to limitations of the
+codegen backend:
+
+- Can only be in the signature of a function if it is annotated with the
+  appropriate target feature (see [*ABI*][abi])
+
+- Cannot be stored in compound types (structs, enums, etc)
+
+    - Including coroutines, so these types cannot be held across an await
+      boundary in async functions
+
+    - `repr(transparent)` newtypes could be permitted with scalable vectors
+
+    - **Exception:** Scalable vectors can be stored in arrays
+
+    - **Exception:** Scalable vectors can be stored in structs with every
+      element of the same type (but only if that struct is annotated with
+      `#[rustc_scalable_vector]`)
+
+- Cannot be used in arrays
```

Comment:
you state they can be stored in arrays as an exception, but here you contradict that.

Reply:
Fixed in 442beeae4054837c76635636b4a7f7da3b3c7802